### PR TITLE
86 storage space and time estimation

### LINKED ISSUE #86: Storage space and time estimation
It would be helpful to get the needed disk space when saving large parameter variation. Then an error can also be thrown when not enough free space is in the directory. It would also be nice to get an estimation of the needed time to simulate large parameter variations.

### PR DESCRIPTION
**Pull request by HvanderStok**
Closes #86

### CODE REVIEW

File: ebcpy/simulationapi/__init__.py [hunk 1]
```diff
@@ -328,25 +331,61 @@ def simulate(self,
...
                 results.append(result)
-                self._log_simulation_process()
+                self._n_sim_counter += 1
+                if self._n_sim_counter == self.n_cpu:
+                    t1 = time.time()
+                if self._n_sim_counter > self.n_cpu:
+                    self._remaining_time(t1)
+                if self._n_sim_counter == 1 and return_option == 'savepath':
+                    self._check_disk_space(result)
+            print(" ")
```
Comment (FWuellhorst) [inline ebcpy/simulationapi/__init__.py:341]: Why is the print necessary? Print should be avoided in pypi packages.

Ok, I see it further down below. Is it possible to do the same for logging as well?

File: ebcpy/simulationapi/__init__.py [hunk 2]
```diff
@@ -328,25 +331,61 @@ def simulate(self,
...
-        progress = int(self._n_sim_counter / self._n_sim_total * 100)
-        if progress == self._progress_int + 10:
-            if self.logger.isEnabledFor(level=logging.INFO):
-                self.logger.info(f"Finished {progress} % of all {self._n_sim_total} simulations")
-            self._progress_int = progress
+    def _remaining_time(self, t1):
+        """
+        Helper function to calculate the remaining simulation time and log the finished simulations.
+        The function can first be used when a simulation has finished on each used cpu, so that the
+        start-up time is not considered in the time estimation.
```
Comment (FWuellhorst) [inline ebcpy/simulationapi/__init__.py:358]: start-up time (e.g. translation)

File: ebcpy/simulationapi/__init__.py [hunk 3]
```diff
@@ -328,25 +331,61 @@ def simulate(self,
...
+
+        :param float t1:
+            Start time after n_cpu simulations.
```
Comment (FWuellhorst) [inline ebcpy/simulationapi/__init__.py:361]: Depending on the case, the first 10 simulations on 10 cores may be executed on e.g. 6 cores only, if the simulation is really fast for instance. 
Please note, that this is just an assumption. (in line 338 as a comment)

File: ebcpy/simulationapi/__init__.py [hunk 4]
```diff
@@ -328,25 +331,61 @@ def simulate(self,
...
+        t_remaining = (time.time() - t1)/(self._n_sim_counter-self.n_cpu) * (self._n_sim_total - self._n_sim_counter)
+        p_finished = self._n_sim_counter/self._n_sim_total * 100
+        print(f"\rFinished {np.round(p_finished, 1)} %. "
+              f"Approximately remaining time: {timedelta(seconds=int(t_remaining))} ", end="")
+
+    def _check_disk_space(self, filepath):
+        """
+        Checks how much disk space all simulations will need on a hard drive
+        and throws a warning when not enough space is free.
+        Works only for multiprocessing.
```
Comment (FWuellhorst) [inline ebcpy/simulationapi/__init__.py:372]: Note the 5 % assumption in the doc

**Review by FWuellhorst — CHANGES_REQUESTED** [review]
Only minor doc string changes requested

File: ebcpy/simulationapi/__init__.py (hunk 1 above)
Comment (HvanderStok) [reply ebcpy/simulationapi/__init__.py]: I have changed the `print` statements to `sys.stderr.write`. Is this maybe better? Unfortunately, it is in my opinion not nicely possible to achieve it with the logging module.

**Review by FWuellhorst — APPROVED** [review]
Thanks for the nice feature and work!

**Pull request merged**